client/actions/auth/index.js

```diff
@@ -29,6 +30,24 @@ export default class AuthActionCreator {
     }
   }
 
+  static doLogin (email, password) {
+    return async (dispatch, getState) => {
+      try {
+        const state = getState()
+        const apiOptions = apiOptionsFromState(state)
+        const user = await usersApiClient.login(apiOptions, email, password)
+        if (!user.hasOwnProperty('error')) {
+          dispatch(AuthActionCreator.login(user))
+          dispatch(push('/'))
+        } else {
+          dispatch({ type: LOGIN_FAILED, error: user.error })
```

Comment:
check [here](https://github.com/acdlite/flux-standard-action#actions) for how I was thinking about formatting actions

the `localStorageMiddleware`, `promiseMiddleware`,  and project actions and reducer were set up to use this format